src/test/java/factoid/converter/FactoidToBiopaxTest.java

```diff
@@ -50,6 +50,426 @@ public void testToOwl() throws IOException {
       .flatMap(Set::stream).collect(Collectors.toList());
     assertThat(controllers.stream().filter(SmallMolecule.class::isInstance).count(), is(2L));
   }
+  
+  @Test
+  public void testMolecularInteraction() throws IOException {
+	  String templates = "[{\n" +
+		  "    \"type\": \"Molecular Interaction\",\n" +
+		  "    \"participants\": [\n" +
+		  "      {\n" +
+		  "        \"type\": \"protein\",\n" +
+		  "        \"name\": \"BMP2\",\n" +
+		  "        \"xref\": {\n" +
+		  "          \"id\": P12643,\n" +
+		  "          \"db\": \"uniprot\"\n" +
+		  "        }\n" +
+		  "      },\n" +
+		  "      {\n" +
+		  "        \"type\": \"protein\",\n" +
+		  "        \"name\": \"LEP\",\n" +
+		  "        \"xref\": {\n" +
+		  "          \"id\": \"P41159\",\n" +
+		  "          \"db\": \"uniprot\"\n" +
+		  "        }\n" +
+		  "      }\n" +
+		  "    ]\n" +
+		  "  }]";
+
+	  FactoidToBiopax converter = new FactoidToBiopax();
+	  converter.addToModel(templates); //processing
+
+	  Model m = converterResultToModel(converter.convertToBiopax());
+	  assertThat(m.getObjects().size(), equalTo(7));
+
+	  Set<RelationshipXref> xrefs = m.getObjects(RelationshipXref.class);
+	  assertThat(xrefs, notNullValue());
+	  assertThat(xrefs.size(), equalTo(2));
+	  xrefs.stream().forEach(x -> {
+		  assertThat(x.getDb(), equalTo("uniprot"));
+		  assertThat(x.getId(), isOneOf("P41159","P12643"));
+	  });
+	  
+	  Set<MolecularInteraction> intns = m.getObjects(MolecularInteraction.class);
+	  assertThat(intns, not(empty()));
+	  MolecularInteraction intn = intns.iterator().next();
+	  Set<Entity> participants = intn.getParticipant();
+
+	  assertThat(participants.size(), equalTo(2));
+	  participants.stream().forEach(p -> {
+		  assertThat(p, instanceOf(Protein.class));
+		  assertThat(((Protein)p).getEntityReference(), notNullValue());
+		  assertThat(((Protein)p).getEntityReference().getXref(), not(empty()));
+	  });
+  }
+  
+  @Test
+  public void testExpressionRegulation() throws IOException {
+	  String templates = "[{\n" +
```

Comment:
Why do you use such a weird example (it's very confusing, not from test2.json where JUN inhibits the transcription/translation of BMP2, and contains modification field)? It would be much easier to undestand and test if you'd simply copy/pasted the objects from test2.json and added more, but only if makes sens, unless you're testing illegal data entries (which should not occur because Factoid cannot generate those...)